packages/logging/package.json

```diff
@@ -22,7 +22,8 @@
     "@mongosh/history": "2.4.6",
     "@mongosh/types": "3.6.0",
     "mongodb-log-writer": "^2.3.1",
-    "mongodb-redact": "^1.1.5"
+    "mongodb-redact": "^1.1.5",
+    "node-machine-id": "^1.1.12"
```

Comment:
This is an unmaintained package which always runs multiple child processes to get its result, I do wonder if there are better alternatives out there, even if it's just hashing `os.hostname()` and maybe the mac addresses reported in `os.networkInterfaces()`?

Reply:
yeah I agree, I wasn't sure about the package though it is used by some pretty big projects and with no good alternatives. If we go in a different direction we'd likely want the Atlas CLI to adopt the same as well but considering the library talks about lack of reliability of Mac addresses etc., seem like it won't be that reliable.

I'm wondering if we could instead do a shared mongodb devtools like `~/.mongodb-tools` directory or something where we write a new UUID for anonymousId and let mongosh, Compass, and Atlas CLI read and use this if it exists already. Seems like hoping all tools can reliably com up with the same device identifier is more unreliable anyhow.

Reply:
haha just realized the "big projects which use this" includes https://www.npmjs.com/package/realm?activeTab=dependencies though it seems like it's just used in an example project 😅

Reply:
> I'm wondering if we could instead do a shared mongodb devtools like `~/.mongodb-tools/config.yaml` directory or something where we write a new UUID for anonymousId and let mongosh, Compass, and Atlas CLI read and use this if it exists already.

If that is an option, we should definitely go with it. I had assumed we'd want a true "device" ID, not something that would be per-user, but in today's world the latter is probably at least as good as the former. (It also has the advantage of the user being able to control it, if necessary.)

> Seems like hoping all tools can reliably come up with the same device identifier is more unreliable anyhow.

Agreed 👍